aspnetcore/signalr/messagepackhubprotocol.md

```diff
@@ -0,0 +1,93 @@
+---
+title: Use MessagePack Hub Protocol in SignalR for ASP.NET Core
+author: 
+description: Learn how to add MessagePack Hub Protocol to ASP.NET Core SignalR.
+manager: wpickett
+monikerRange: '>= aspnetcore-2.1'
+ms.author: 
+ms.custom: mvc
+ms.date: 06/04/2018
+ms.prod: aspnet-core
+ms.technology: aspnet
+ms.topic: article
+uid: signalr/messagepack
+---
+
+# Use MessagePack Hub Protocol in SignalR for ASP.NET Core
+
+This article assumes the reader is familiar with the topics covered in [Get Started](xref:signalr/get-started).
+
+## What is MessagePack
+
+MessagePack is a binary serialization format that is fast and compact. It is useful when performance and bandwidth are a concern because it creates smaller packets compared to JSON. Because it is a binary format, it will be unreadable when looking at network traces and logs unless you pass the bytes through a MessagePack parser. You can see more information about the format at https://msgpack.org/index.html. SignalR has added built-in support for the MessagePack format and added APIs for the client and server to use.
+
+## Configure MessagePack on the Server
+
+To enable the MessagePack Hub Protocol on the server you will first need to install the `Microsoft.AspNetCore.SignalR.Protocols.MessagePack` package in your app. Next add `.AddMessagePackProtocol()` to the  `AddSignalR()` call and now the server will support MessagePack.
+
+> [!NOTE]
+> Json is supported by default, adding MessagePack is addative so both will be supported at the same time.
+
+```csharp
+services.AddSignalR()
+    .AddMessagePackProtocol();
+```
+
+If you want to customize how MessagePack will format your data `.AddMessagePackProtocol()` takes a func for configuring options. A property named `FormatterResolvers` is exposed in the options and this is a list of resolvers that are used by MessagePack to determine the format your messages will be in. For more information on how the resolvers work visit the MessagePack library we use at [MessagePack-CSharp](https://github.com/neuecc/MessagePack-CSharp). Or you can use attributes on the objects you want to serialize to define how they should be handled.
```

Comment:
If you want to customize how MessagePack will format your data `.AddMessagePackProtocol()` takes a func for configuring options. A property named `FormatterResolvers` is exposed in the options and this is a list of resolvers that are used by MessagePack to determine the format your messages will be in.
->
To customize how MessagePack will format data, `AddMessagePackProtocol` takes a Func for configuring options. A property named `FormatterResolvers` is exposed in the options. The `FormatterResolvers` is a list of resolvers that are used by MessagePack to determine the format messages will be in.

Reply:
For more information on how the resolvers work visit the MessagePack library we use at MessagePack-CSharp. Or you can use attributes on the objects you want to serialize to define how they should be handled.
->
 For more information on how the resolvers work, visit the MessagePack library at [MessagePack-CSharp](https://github.com/neuecc/MessagePack-CSharp). Use attributes on the objects you want to serialize to define how they should be handled.

Reply:
> Use attributes on the objects you want to serialize to define how they should be handled.

This makes it sound like the attributes are required, which they aren't. How should we word-smith this better?

"Or use attributes..."?

Reply:
"Or" is ok but...
-the phrase "you want to serialize" means that it is not required
-"Or" is an alternative to the immediate previous sentence, which is about where to find more information. 

How about this?
Attributes can be used on the objects you want to serialize to define how they should be handled.